test/perf/run_nightly_perf.py

```
@@ -0,0 +1,863 @@
+import time
+from threading import Thread
+import urllib.request
+import yaml
+
+# Global flag to stop the metric monitoring thread
+stop_monitoring = False
+
+def run_cmd(cmd, check=True, capture_output=True, text=True):
+    print(f"Running: {cmd}")
+    if isinstance(cmd, str):
+        res = subprocess.run(cmd, shell=True, check=check, capture_output=capture_output, text=text)
```

Ruff (0.15.21)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)